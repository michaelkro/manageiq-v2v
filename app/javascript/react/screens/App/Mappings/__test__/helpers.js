import uuid from 'uuid/v4';

import {
  TRANSFORMATION_MAPPING_ITEM_SOURCE_TYPES,
  TRANSFORMATION_MAPPING_ITEM_DESTINATION_TYPES
} from '../screens/MappingWizard/MappingWizardConstants';

export const generateMappingItem = (targetProvider, type, source_id, destination_id) => ({
  source_id: source_id || uuid(),
  source_type: TRANSFORMATION_MAPPING_ITEM_SOURCE_TYPES[type],
  destination_id: destination_id || uuid(),
  destination_type: TRANSFORMATION_MAPPING_ITEM_DESTINATION_TYPES[targetProvider][type]
});
