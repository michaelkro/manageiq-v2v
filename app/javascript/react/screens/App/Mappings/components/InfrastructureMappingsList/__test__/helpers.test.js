import { getMappingType, getHeaderText } from '../helpers';
import { generateMappingItem } from '../../../__test__/helpers';

const osp_mapping_items = [
  generateMappingItem('openstack', 'cluster'),
  generateMappingItem('openstack', 'datastore'),
  generateMappingItem('openstack', 'network')
];

const rhv_mapping_items = [
  generateMappingItem('rhevm', 'cluster'),
  generateMappingItem('rhevm', 'datastore'),
  generateMappingItem('rhevm', 'network')
];

describe('getMappingType', () => {
  test('detects OSP mappings', () => {
    expect(getMappingType(osp_mapping_items)).toBe('openstack');
  });

  test('detects RHV mappings', () => {
    expect(getMappingType(rhv_mapping_items)).toBe('rhevm');
  });
});

describe('getHeaderText', () => {
  test('formats header text for OSP mappings', () => {
    expect(getHeaderText(osp_mapping_items)).toMatchSnapshot();
  });

  test('formats header text for RHV mappings', () => {
    expect(getHeaderText(rhv_mapping_items)).toMatchSnapshot();
  });
});
