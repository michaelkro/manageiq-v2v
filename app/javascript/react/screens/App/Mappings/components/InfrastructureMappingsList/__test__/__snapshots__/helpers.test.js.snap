// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getHeaderText formats header text for OSP mappings 1`] = `
Object {
  "sourceClusters": "Source Provider \\\\ Datacenter \\\\ Cluster",
  "sourceDatastores": "Source Provider \\\\ Datacenter \\\\ Cluster \\\\ Datastore",
  "sourceNetworks": "Source Provider \\\\ Datacenter \\\\ Cluster \\\\ Network",
  "targetClusters": "Target Provider \\\\ Project",
  "targetDatastores": "Target Provider \\\\ Project \\\\ Volume Type",
  "targetNetworks": "Target Provider \\\\ Project \\\\ Network",
}
`;

exports[`getHeaderText formats header text for RHV mappings 1`] = `
Object {
  "sourceClusters": "Source Clusters",
  "sourceDatastores": "Source Datastores",
  "sourceNetworks": "Source Networks",
  "targetClusters": "Target Clusters",
  "targetDatastores": "Target Datastores",
  "targetNetworks": "Target Networks",
}
`;
